framework/include/materials/MaterialPropertyStorage.h

```diff
@@ -310,28 +329,47 @@ class MaterialPropertyStorage
                  unsigned int side,
                  unsigned int n_qpoints);
 
+  /// Initializes just one hashmap's entries
+  void initProps(MaterialData & material_data,
+                 PropsType & mat_props_map,
+                 const Elem * elem,
+                 unsigned int side,
+                 unsigned int n_qpoints);
+
+  // You'd think a private mutex would work here, so I'll leave this
+  // in the namespace for when that happens, but CI thinks that can
+  // make us crash, so I'll just initialize this to Threads::spin_mtx
```

Comment:
How does CI communicate this? I would certainly prefer a class mutex over the global one

Reply:
Via segfaults, IIRC.  I didn't understand why.  I'd definitely prefer a class mutex too.  Perhaps I used a per-class mutex and a static mutex would behave better?  I'll try that.

Reply:
Looks like we didn't try the static class mutex?

Reply:
Not yet.  I'd like to make the upgrade (that's why the reference is here; originally the global mutex was hardcoded elsewhere), but let me try it out in a separate PR?  This one is important enough and urgent enough and invasive enough that I'd rather not get it mixed up with something that might cause intermittent threading race conditions.